apps/trackers/jira/query.py

```diff
@@ -585,6 +584,105 @@ def generate(self):
         self.generate_embargo_status()
         self.generate_special_handling()
 
+    @cached_property
+    def most_important_affect(self):
+        """
+        Selects the most important affect to use for displaying
+        CVE/CVSS/CWE/Source in Vulnerability issuetype Jira tracker
+        fields, which is relevant for multi-flaw trackers where
+        the tracker can have multiple sets of CVE/CVSS/CWE/source
+        but the tracker should display only one such set.
+        Quote from 2024-09-27 in OSIDB-3348:
+          "For multiflaw trackers, select CVE with highest Impact value,
+           if there are multiple such CVE, select the one where flaw has
+           the oldest created_date/time value. Use CVE/cvss/cwe/source
+           from that CVE."
+        Note: The writer assumed a flaw has only one authoritative CVE
+              object (the one issued by RH) and Flaw == CVE in this context.
+        Note: Assuming validations and administrative procedures ensure
+              that the impact model fields are in accordance with the
+              cvss_scores fields.
+        """
+        affects_impacts = sorted(
+            [
+                (affect, affect.aggregated_impact)
+                for affect in self.tracker.affects.all()
+            ],
+            key=lambda pair: pair[1],
+        )
+        greatest_impact_affects_impacts = [
+            (affect, impact)
+            for affect, impact in affects_impacts
+            if impact == affects_impacts[-1][1]
+        ]
+        flaw_created_dt_affects = sorted(
+            [
+                (affect.flaw.created_dt, affect)
+                for affect, impact in greatest_impact_affects_impacts
+            ],
+            key=lambda pair: pair[0],
+        )
+        most_important_affect = flaw_created_dt_affects[0][1]
+        return most_important_affect
+
+    @cached_property
+    def most_important_cvss(self):
+        """
+        Returns a RH-issued CVSS.
+        For explanation see docstring of most_important_affect.
+        If the most important affect doesn't have a related RH CVSS score,
+        then no other affect's/flaw's CVSS score is selected even for multi-flaw
+        trackers, so that the set of CVE/CVSS/CWE/Source is consistent.
+        """
+
+        affect = self.most_important_affect
+        if affect.cvss_scores.filter(issuer=AffectCVSS.CVSSIssuer.REDHAT).exists():
+            # Affect override present.
+            return (
+                affect.cvss_scores.filter(issuer=AffectCVSS.CVSSIssuer.REDHAT)
+                .order_by("-version")
+                .first()
+            )
+
+        return (
+            affect.flaw.cvss_scores.filter(issuer=FlawCVSS.CVSSIssuer.REDHAT)
+            .order_by("-version", "-score")
```

Comment:
Why do we sort by `score` here on Flaw model but not above in the Affect model?

Reply:
Because I didn't realize there's already validation that ensures that there can't be multiple CVSS scores with a given issuer and version for a given Flaw/Affect, and that the CVSS test factories already ensure that too. And because in a previous quite different WIP version of the tests and algorithm I somehow came up with this sorting for some reason and forgot to treat both AffectCVSS and FlawCVSS in the same way. Removed the sorting by score. Thank you for noticing.